src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1603,112 +1646,98 @@ public RefreshItem(String keyInCache, AuthenticationRequest request, TokenCacheI
             mKey = keyInCache;
             mMultiResource = item.getIsMultiResourceRefreshToken();
             mRefreshToken = item.getRefreshToken();
+            mFamilyToken = !StringExtensions.IsNullOrBlank(item.getFamilyClientId());
             mUserInfo = item.getUserInfo();
             mRawIdToken = item.getRawIdToken();
             mTenantId = item.getTenantId();
-            final UserInfo userInfo = item.getUserInfo();
-            if (userInfo != null) {
-                mKeyWithUserId = CacheKey.createCacheKey(request, userInfo.getUserId());
-                mKeyWithDisplayableId = CacheKey.createCacheKey(request, userInfo.getDisplayableId());
-            }
         }
     }
 
-    private RefreshItem getRefreshToken(final AuthenticationRequest request, boolean useMultiResourceRefreshToken) {
+    private RefreshItem getRefreshToken(final AuthenticationRequest request) {
```

Comment:
I would recommend against having a method that could retrieve multiple types of refresh tokens. It makes the logic a lot harder to follow because you're not sure in the calling method what exactly you're getting.